dev/lib/syntax.js

```diff
@@ -176,11 +176,24 @@ function tokenizeTable(effects, ok, nok) {
 
   /** @type {State} */
   function start(code) {
+    const { containerState } = self;
+    const prevRowCount = containerState.rowCount ?? 0;
+    const hasDelimiterRow = containerState.hasDelimiterRow ?? false;
+
     // @ts-expect-error Custom.
     effects.enter('table')._align = align
     effects.enter('tableHead')
     effects.enter('tableRow')
 
+    // increment row count
+    const crrRowCount = prevRowCount + 1;
+    containerState.rowCount = crrRowCount;
+
+    // Max 2 rows processing before delimiter row
+    if (hasDelimiterRow || crrRowCount > 2) {
```

Comment:
All tests will pass if you change the condition to `hasDelimiterRow || crrRowCount > 3`.

Because the given markdown int the test:

```markdown
| a |
| - |
heading
-
```

needs 3-rows processing.

However, Is this check required for the implementation of common-mark or GFM? I think it'll be not bad to omit the test if it is a corner case.